lib/elixir/lib/macro/env.ex

```diff
@@ -555,8 +555,17 @@ defmodule Macro.Env do
 
   ## Options
 
-    * `:allow_locals` - when set to `false`, it does not attempt to capture
-      local macros defined in the current module in `env`
+    * `:allow_locals` - controls how local macros are resolved.
+      Defaults to `true`.
+      
+      - When `false`, does not attempt to capture local macros defined in the
+        current module in `env`
+      - When `true`, uses a default resolver that looks for public macros in
+        the current module
+      - When a function, uses the function as a custom local resolver. The function
+        must have the signature: `(meta, name, arity, kinds, env) -> function() | false`
```

Comment:
We don't really need the `kinds`, do we? I think it can be hidden from the API so we keep this as a 4 arity function?

Reply:
We could hide kinds and assume it's a macro resolver.

Reply:
Let’s do that, since it is for expansion (so it has to be macros)

Reply:
done